Add information on plugin-versioning (#615)
tableau/connector-plugin-sdk: docs/design.md

You might find other superclass values in the workbook XML from an existing Tableau connection, but we don't recommend using them. They haven't been tested.

## Assign a plugin version

Plugin-version is a requirement for submitting your connector to the Tableau Connector Gallery.  It helps customers and partners understand which version of a connector is being used currently and if it is the most up-to-date. 
You should start your plugin version with `plugin-version='1.0.0'` and increment it based on major or minor changes to your connector. 

## Know the min-tableau-version

> For the min-tableau-version, is checked before a connector is registered to tableau. If the current version of tableau does is less than the min-tableau-version then the connector is not registered. 

Reviewer: ```suggestion
For the min-tableau-version, is checked before a connector is loaded in Tableau. If the current version of Tableau is less than the min-tableau-version then the connector is not loaded. 
```

You might find other superclass values in the workbook XML from an existing Tableau connection, but we don't recommend using them. They haven't been tested.

## Assign a plugin version

> Plugin-version is a requirement for submitting your connector to the Tableau Connector Gallery, especially with a partner releasing a newer version in Tableau gallery and notifying customers to download an update. 

Reviewer: ```suggestion
Plugin-version is a requirement for submitting your connector to the Tableau Connector Gallery.  It helps customers and partners understand which version of a connector is being used currently and if it is the most up-to-date. 
```

Author: Made the required change to it.

You should start your plugin version with `plugin-version='1.0.0'` and increment it based on major or minor changes to your connector. 

> ## Set your min-tableau-version

Reviewer 2: People don't set this, the packager automatically sets this. Letting them know it exists is fine, but anything they put here will be overwritten by the packager.

Author: Updated the part about min-tableau-version being set by the packager.

For the min-tableau-version, is checked before a connector is registered to tableau. If the current version of tableau does is less than the min-tableau-version then the connector is not registered. 
You should set your `min-version-tableau` to the version that has the features used in your connector. 

***Note: `version` in the manifest file does not refer to plugin version. Please leave this set to `version='18.1'`*** <br />

***FAQs about Versioning***

> - When should I update I update min-tableau-version?

Reviewer 2: They shouldn't update min-tableau-version, see previous comment

Author: Removed this part.

    - Update the min-tableau version only when there is a breaking change dependent on a feature on the newer version of Tableau.  

- What should I (partner) do if I need to make a breaking change to the connector?

>   - If there is a breaking change to your connector you should submit a new connector. This will ensure that there is a connector for the workbooks that used older connector and prevents the downgrade issue with server and desktop using a different version of the connectors.

Reviewer: ```suggestion
  - If there is a breaking change to your connector you should submit a new connector. This will ensure that there is a connector for an existing workbook that used the older connector. It also prevents issues with workbook version downgrade scenarios when Tableau Server and Desktop are using different version of a connector.
```

Author: Thanks, Ross. I made the suggested change to this.

## Assign a plugin version

Plugin-version is a requirement for submitting your connector to the Tableau Connector Gallery.  It helps customers and partners understand which version of a connector is being used currently and if it is the most up-to-date. 
You should start your plugin version with `plugin-version='1.0.0'` and increment it based on major or minor changes to your connector. 

## Know the min-tableau-version
For the min-tableau-version, is checked before a connector is registered to tableau. If the current version of tableau does is less than the min-tableau-version then the connector is not registered. 

> This is set by the packager and there is no need set this manually. 

Reviewer 2: ```suggestion
This is set by the packager and if set manually will be overwritten.
```
